goodtablesio/tasks.py

```diff
@@ -17,61 +17,19 @@
 
 
 @app.task(name='goodtableio.tasks.validate')
-def validate(payload):
+def validate(task_desc):
     """Main validation task.
 
     Args:
-        payload (dict): task payload
+        task_desc (dict): task descriptor
 
-    Payload structure reflects `goodtables.Inspector` API:
-        - `config` key will be used as `Inspector` options
-        - all other keys will be used as `Inspector.inspect` options
-
-    Examples:
-        - table preset:
-            ```
-            payload = {
-                'source': 'path.csv',
-                'config': {
-                    'checks': 'structure',
-                }
-            }
-            ```
-        - tables preset:
-            ```
-            payload = {
-                'source': [
-                    {'source': 'path1.csv'},
-                    {
-                        'source': 'path2.csv',
-                        'schema': 'schema.json',
-                        'delimiter': ';',
-                    }
-                ],
-                'preset': 'tables',
-                'config': {
-                    'checks': 'schema',
-                    'error_limit': 10,
-                    'order_fields': True,
-                }
-            }
-            ```
-        - datapackage preset:
-            ```
-            payload = {
-                'source': 'datapackage.json',
-                'preset': 'datapackage',
-                'config': {
-                    'table_limit': 2,
-                }
-            }
-            ```
+    See `schemas/task-desc.yml`.
 
     """
 
     # Get report
-    inspector = Inspector(**payload.pop('config', {}))
-    report = inspector.inspect(**payload)
+    inspector = Inspector(**task_desc['settings'])
+    report = inspector.inspect(task_desc['files'], preset='tables')
```

Comment:
Yes.

I've tried to make very similiar configuration (yml) and descriptor (api) formats. And proposed by @pwalsh yml can't support datapackages because json also could be just a tabular file.

So I suppose in both formats we should introduce something like `datapackage` or `datapackages` attribute wich is mutually exclusive with `files`:
```
datapackages: *
settings:
  ...
```